### PR TITLE
 Added support for Content Security Policy, Rebase #330

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
This is a rebase of #330

File: debug_toolbar/templates/debug_toolbar/panels/templates.html [hunk 1]
```diff
@@ -35,7 +35,7 @@
 	<dt><strong>{{ key|escape }}</strong></dt>
 	<dd>
 		<div class="djTemplateShowContextDiv"><a class="djTemplateShowContext"><span class="toggleArrow">&#x25B6;</span> {% trans "Toggle context" %}</a></div>
-		<div class="djTemplateHideContextDiv" style="display:none;"><code>{{ value|escape }}</code></div>
+		<div class="djTemplateHideContextDiv" hidden><code>{{ value|escape }}</code></div>
```
Comment (tim-schilling) [inline debug_toolbar/templates/debug_toolbar/panels/templates.html]: Is there a reason why you didn't use ```hidden="hidden"``` here and on line 22?

**Comment by tim-schilling** [comment]
This looks pretty good. Could you please add a ```djdt-``` prefix to the ```width-*``` classes? Developers may have stylesheets defined that would apply their styles to elements with that class name. Adding that prefix would help mitigate that possibility.

**Comment by graingert** [comment]
Done, I've also prefixed the CSS class 'highlighted'

**Comment by thedrow** [comment]
@graingert Can you please rebase again? The PR contains merge conflicts.

**Comment by graingert** [comment]
re-based, also killed that other "hidden"

**Comment by graingert** [comment]
@tim-schilling @thedrow poke

**Comment by tim-schilling** [comment]
I tried setting up a test application with a content security policy enabled and I had to add ```'self' ajax.googleapis.com``` to the ```script-src``` directive and ```'self' data:``` to the ```img-src``` directive. Does that sound correct? If so, then I'm going to add some documentation that explains this.

**Comment by graingert** [comment]
Yes that's correct. Although if you set a custom JQuery URL it will be whatever the origin of that URL is.

**Comment by tim-schilling** [comment]
I was doing some more testing and whenever I click on the SQL tab, it generates the unsafe-eval error. Can you duplicate that issue?

**Comment by graingert** [comment]
@tim-schilling Can you paste me the full error?

Do you have your test project on GitHub?

**Comment by graingert** [comment]
I've repeated the issue. We need to allow `script-src "unsafe-eval"` for now.

For now can you merge this and I will work on a fix?

**Comment by aaugustin** [comment]
Yes, I think we should merge this PR before it needs rebasing again.

@graingert Fortunately there isn't a lot of activity, only @tim-schilling and myself merge patches.

I think it would be helpful to add guidelines to the contributing documentation so future changes don't break CSP inadventently. I'm not sure if it could be tested automatically e.g. with Selenium?

@tim-schilling As far as I am concerned, I don't mind merging the patch as is and iterating.

**Comment by tim-schilling** [comment]
I meant to respond earlier. I decided that I'd merge it but not document it
until the unsafe-eval directive isn't needed. I don't think we can say the
toolbar supports a content security policy until that directive is no
longer needed.

**Comment by tim-schilling** [comment]
I rebased your changes on master, and merged them in with commit 8b29127ffb385db22fc2a3296b9904dd73180e56. Thank you for the PR @graingert!

**Pull request closed without merging**